src/Deserialization.h

```diff
@@ -0,0 +1,10 @@
+#ifndef HALIDE_DESERIALIZATION_H
+#define HALIDE_DESERIALIZATION_H
+
+#include "Pipeline.h"
+
+namespace Halide {
+Pipeline deserialize_pipeline(const std::string &filename, const std::unordered_map<std::string, Internal::Parameter> &params);
```

Comment:
Does this need any comments about argument restrictions, error conditions, etc? (Not sure, this is a genuine question)